boilerplate/_lib/boilerplate.mk

```
@@ -1,3 +1,11 @@
+# When running in a CI environment, we usually can't write to the default go cache of
+# ${HOME}/.cache/go-build. This file is the first chunk of Makefile loaded from a repo's `generated-includes.mk`,
+# therefore we compute this upfront and then check it in the various conventions boilerplate supports.
+ifdef PROW_JOB_ID
```

Can we just check if `/tmp` is writable?
```suggestion
if [ -w "/tmp" ]; then
```

---

Are you suggesting that if `/tmp` is writeable we should use it for the cache 100% of the time?

The intent with this change is to:

- Let existing `GOCACHE` behavior work in local dev, so users existing `$HOME/.cache/go-build` can be used, versus all `make` targets re-pulling dependencies every time
- In CI environments, if `go env GOCACHE` returns `off`, that means the default directory `$HOME/.cache/go-build` is not writeable, so this was just a way to always ensure that is writeable up front without checking it in all the various conventions

I played with a few iterations but the best I could come up with was just keying off CI vs local to handle both (ignoring a users local GOCACHE value being broken). Open to other options, but I don't think just using `/tmp` if writeable solves it all.

---

@dustman9000 I have made some changes, which also allowed the existing tests to pass (which are proving very difficult to use as they fail to even run locally).